jooby-filewatcher/src/main/java/org/jooby/filewatcher/FileEventOptions.java

```diff
@@ -0,0 +1,163 @@
+
+    @Override
+    public boolean matches(final Path path) {
+      return true;
+    }
+
+    @Override
+    public String toString() {
+      return "**/*.*";
+    }
+  };
+
+  private static final WatchEvent.Kind<?>[] KINDS = {ENTRY_CREATE, ENTRY_DELETE, ENTRY_MODIFY };
+
+  private List<PathMatcher> matchers = new ArrayList<>();
+
+  private Modifier modifier = SensitivityWatchEventModifier.HIGH;
+
+  private boolean recursive = true;
+
+  private Class<? extends FileEventHandler> handler;
+
+  private Path path;
+
+  private FileEventHandler handlerInstance;
+
+  FileEventOptions(final Path path, final Class<? extends FileEventHandler> handler)
+      throws IOException {
+    if (!Files.exists(path)) {
+      Files.createDirectories(path);
+    }
+    this.path = path;
+    this.handler = handler;
+  }
+
+  FileEventOptions(final Path path, final FileEventHandler handler) throws IOException {
+    this(path, handler.getClass());
+    this.handlerInstance = handler;
+  }
+
+  /**
+   * Append a kind filter.
+   *
+   * The default filter is: {@link StandardWatchEventKinds#ENTRY_CREATE},
+   * {@link StandardWatchEventKinds#ENTRY_DELETE} and {@link StandardWatchEventKinds#ENTRY_MODIFY}.
+   *
+   * @param kind Filter type.
+   * @return This options.
+   */
+  public FileEventOptions kind(final WatchEvent.Kind<Path> kind) {
+    requireNonNull(kind, "WatchEvent.Kind required.");
+    kinds.add(kind);
+    return this;
+  }
+
+  /**
+   * Turn on/off watching of subfolder.
+   *
+   * @param recursive True to watch on subfolder.
+   * @return This options.
+   */
+  public FileEventOptions recursive(final boolean recursive) {
+    this.recursive = recursive;
+    return this;
+  }
+
+  /**
+   * Add a path filter using <code>glob</code> expression, like <code><pre>** / *.java</pre><code>,
+   * etc...
+   *
+   * @param expression Glob expression.
+   * @return This options.
+   */
+  public FileEventOptions includes(final String expression) {
+    requireNonNull(expression, "Glob expression required.");
+    this.matchers.add(new GlobPathMatcher(expression));
+    return this;
+  }
+
+  /**
+   * Set a watch modifier. Default is: <code>HIGH</code>.
+   *
+   * @param modifier Watch modifier.
+   * @return This options.
+   */
+  public FileEventOptions modifier(final WatchEvent.Modifier modifier) {
+    requireNonNull(modifier, "Modifier required.");
+    this.modifier = modifier;
+    return this;
+  }
+
+  Modifier modifier() {
+    return modifier;
+  }
+
+  WatchEvent.Kind<?>[] kinds() {
+    return kinds.size() == 0 ? KINDS : kinds.toArray(new WatchEvent.Kind[0]);
+  }
+
+  PathMatcher filter() {
+    return matchers.size() == 0 ? TRUE : new FirstOfPathMatcher(matchers);
+  }
+
+  boolean recursive() {
+    return recursive;
+  }
+
+  Path path() {
+    return path;
+  }
+
+  FileEventHandler handler(
+      final Function<Class<? extends FileEventHandler>, FileEventHandler> factory) {
```

Comment:
4 spaces used for indent instead of 2.
Is this an exception when breaking long lines in method parameters?